helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -412,6 +430,61 @@ private void refreshBaseline(ResourceControllerDataProvider clusterData,
     return newAssignment;
   }
 
+  private void asyncReportBaselineDivergenceGauge(ResourceControllerDataProvider clusterData,
+      Map<String, ResourceAssignment> baseline,
+      Map<String, ResourceAssignment> bestPossibleAssignment) {
+    asyncExecute(clusterData.getAsyncTasksThreadPool(), () -> {
+      try {
+        reportBaselineDivergenceGauge(baseline, bestPossibleAssignment);
+      } catch (Exception e) {
+        LOG.error("Failed to report BaselineDivergenceGauge metric.", e);
+      }
+      return null;
+    });
+  }
+
+  private void reportBaselineDivergenceGauge(Map<String, ResourceAssignment> baseline,
+      Map<String, ResourceAssignment> bestPossibleAssignment) {
+    double baselineDivergence = measureBaselineDivergence(baseline, bestPossibleAssignment);
+
+    BaselineDivergenceGauge baselineDivergenceGauge = _metricCollector.getMetric(
+        WagedRebalancerMetricCollector.WagedRebalancerMetricNames.BaselineDivergenceGauge.name(),
+        BaselineDivergenceGauge.class);
+
+    baselineDivergenceGauge.updateValue(baselineDivergence);
+  }
+
+  private double measureBaselineDivergence(Map<String, ResourceAssignment> baseline,
```

Comment:
Can you high-levelly describe what's the logic here? I cannot easily follow the logic. And I do expect to see some state/instance compare, but there is nothing like that. So not sure if this is the expected logic or not.

Reply:
This is at partition level.

Reply:
+1. Some description would be very helpful.